### ISSUE TITLE
Bug - dropdown - accessibility issues

### ISSUE THREAD

**Issue by nikkimk** [patternfly/patternfly#6560]
**Describe the problem**
This is an issue with 
- [x] Patternfly 5  
- [ ] Patternfly 6  
- [x] other Patternfly 4

Based on a review by @adamjohnson and me, we have determined that dropdown has a number of accessibility issues, including:
- Missing `aria-controls` (v4 and v5)
- Missing `aria-haspopup` (v5)
- Select examples should be a comboxbox role with single-select pattern (v4)

**How do you reproduce the problem?**

***Version 4***
1. Go to [Basic dropdowns](https://v4-archive.patternfly.org/v4/components/dropdown#basic-dropdowns) example.
2. Note that the toggle button does not have an `aria-controls` property with IDREF to the popup menu it expands.
3. Additionally go to [Split button with toggle text example](https://v4-archive.patternfly.org/v4/components/dropdown#split-button-checkbox-with-toggle-text).
4. Note that this example should really follow the [select-only combobox pattern](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/).

***Version 5***
1. Go to [Basic dropdowns](https://www.patternfly.org/components/menus/dropdown#basic-dropdowns) example.
2. Note that the toggle button does not have an `aria-controls` property with IDREF to the popup menu it expands.
3. Note that the toggle button does not have an `aria-haspopup` property.

**Expected behavior**
See the following examples from the from WAI ARIA Authoring Practices Guide for correct use:
- [Navigation menu button example](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/) illustrates use of `aria-controls` and `aria-haspopup`
- [Select-only combobox pattern](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/) illustrates how to implement a dropdown with selections

**Is this issue blocking you?**
No.

**Any other information?**
This creates accessibility issues wherever implemented.

cc/ @markcaron

**Comment by thatblindgeye** [comment]
Going to close this as not planned.

- For the aria attributes, https://github.com/patternfly/patternfly/issues/6561 should cover it as well as the React followups associated to it
- We don't plan to make these updates for v4 since it is 2 versions back, and we can look into making this update for v5 if it's requested from consumers
- For the Dropdown component, the role should still remain as a menu rather than a listbox since the popup content is still only a menu. The split toggle examples don't contain any selectable options, rather just a checkbox in addition to a normal dropdown.
